README updates (#923)
Joystream/joystream: node/README.md

## Joystream-Node - full node/validator

![ Nodes for Joystream](./validator-node-banner.svg)

> joystream-node is the main server application that connects to the network, synchronizes the blockchain with other nodes and produces blocks if configured as a validator node.

Reviewer: ```suggestion
The joystream-node is the main server application that connects to the network, synchronizes the blockchain with other nodes and produces blocks if configured as a validator node.
```

Reviewer 2: Isn't this correct?

Author: If you are referring to the grammar, produces is correct I believe. But the capitalization at the start is correct.

> To setup a full node and validator review the [advanced guide from the helpdesk](https://github.com/Joystream/helpdesk/tree/master/roles/validators).

Reviewer: ```suggestion
A step by step guide to setup a full node and validator on the Joystream testnet, can be found [here](https://github.com/Joystream/helpdesk/tree/master/roles/validators).
```

> ### Pre-built Binaries

Reviewer 2: ```suggestion
### Pre-built binaries
```

The latest pre-built binaries can be downloaded from the [releases](https://github.com/Joystream/joystream/releases) page.

> Generally these will be built from master branch and relevant to the current active testnet.

Reviewer 2: ```suggestion
Generally these will be built from `master` branch and will pertain to the currently active testnet.
```

### Building from source

Clone the repository and install build tools:

```bash
git clone https://github.com/Joystream/joystream.git

cd joystream/

./setup.sh
```

> compile the node and runtime

Reviewer 2: ```suggestion
Compile the node and runtime:
```

```bash
cargo build --release
```

This produces the binary in `./target/release/joystream-node`

> ### Running Local development chain

Reviewer 2: ```suggestion
### Running local development chain
```

```bash
cargo run --release -- --dev
```

If you repeatedly need to restart a new chain,

> This script will build and run a fresh new local development chain purging existing chain data:

Reviewer 2: ```suggestion
this script will build and run a fresh new local development chain (purging existing chain data):
```

```bash
./scripts/run-dev-chain.sh
```

> ### Joystream public testnet

Reviewer 2: ```suggestion
### Joystream Public Testnets
```

> Use the `--chain` argument, and specify the path to the genesis chain.json file for that public network. The joystream public networks are found in [../testnets/](../testnets/)

Reviewer 2: ```suggestion
Use the `--chain` argument, and specify the path to the genesis `chain.json` file for that public network. The JSON "chain spec" files for Joystream public networks can be found in [../testnets/](../testnets/).
```
Also noticed that there only seems to be a `rome.json` in that folder, does it need updating?

Author: If you remember, rome was a new chain, and then the later releases were runtime upgrades and that doesn't require a new chain (genesis) file. I don't think we need to add any other staging or testing networks.

```bash
cargo run --release -- --chain testnets/rome.json
```

### Tests and code quality

> Running unit tests

Reviewer 2: ```suggestion
Running unit tests:
```

```bash
cargo test --all
```

> Running full suite of checks.. tests, formatting and linting:

Reviewer 2: ```suggestion
Running full suite of checks, tests, formatting and linting:
```

```bash
yarn cargo-checks
```

> Always format your rust code with `cargo fmt` before commiting:

Reviewer 2: ```suggestion
Always format your rust code with `cargo fmt` before committing:
```

```bash
cargo fmt --all
```

### Integration tests

```bash
./scripts/run-test-chain.sh
yarn workspace joystream-testing test
```

> To run the integration tests with a different chain, you can omit step running the local development chain and set the node URL using `NODE_URL` environment variable.

Reviewer 2: ```suggestion
To run the integration tests with a different chain, you can omit the step of running the local development chain and simply set the node URL using `NODE_URL` environment variable.
```

### Installing a release build

> If you are building a tagged release from master branch and want to install the executable to your path follow the step below.

Reviewer 2: ```suggestion
If you are building a tagged release from `master` branch and want to install the executable to your path follow the step below.
```

This will install the executable `joystream-node` to your `~/.cargo/bin` folder, which you would normally have in your `$PATH` environment.

```bash
cargo install joystream-node --path node/
```

> Now you can run and connect to the rome testnet:

Reviewer 2: ```suggestion
Now you can run and connect to the Rome testnet:
```
